modules/network/src/test/java/org/apache/ignite/network/internal/AllTypesMessageGenerator.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network.internal;
+
+import java.lang.reflect.Field;
+import java.util.BitSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.ignite.lang.IgniteUuidGenerator;
+import org.apache.ignite.plugin.extensions.communication.MessageCollectionItemType;
+
+/**
+ * Generator for an {@link AllTypesMessage}.
+ */
+public class AllTypesMessageGenerator {
+    /**
+     * Generate a new {@link AllTypesMessage}.
+     *
+     * @param seed Random seed.
+     * @param nestedMsg {@code true} if nested messages should be generated, {@code false} otherwise.
+     * @return Message.
+     * @throws Exception If failed.
+     */
+    public static AllTypesMessage generate(long seed, boolean nestedMsg) {
+        try {
+            var random = new Random(seed);
+
+            var message = new AllTypesMessage();
+
+            Field[] fields = AllTypesMessage.class.getDeclaredFields();
+
+            for (Field field : fields) {
+                field.setAccessible(true);
+
+                TestFieldType annotation = field.getAnnotation(TestFieldType.class);
+
+                if (annotation != null) {
+                    field.set(message, randomValue(random, annotation.value(), nestedMsg));
+                }
+            }
+
+            if (nestedMsg) {
+                Field objectArrayField = AllTypesMessage.class.getDeclaredField("v");
+                objectArrayField.setAccessible(true);
+
+                Field collectionField = AllTypesMessage.class.getDeclaredField("w");
+                collectionField.setAccessible(true);
+
+                Field mapField = AllTypesMessage.class.getDeclaredField("x");
+                mapField.setAccessible(true);
+
+                Object[] array = IntStream.range(0, 10).mapToObj(unused -> generate(seed, false)).toArray();
+
+                objectArrayField.set(message, array);
```

Comment:
I think you got a little bit too excited using reflection here: `message.v = array;`. Same applies to all other field

Reply:
true :D